src/components/v2/appDetails/appDetails.api.ts

```diff
@@ -22,33 +22,29 @@ export const deleteResource = (nodeDetails: any, appDetails: any, envId: string,
     if (!nodeDetails.group) {
         nodeDetails.group = ''
     }
-
-    if (appDetails.appType === AppType.EXTERNAL_HELM_CHART || appDetails.deploymentAppType === DeploymentAppType.helm) {
-        const data = {
-            appId: getAppId(
-                appDetails.clusterId,
-                appDetails.namespace,
-                appDetails.deploymentAppType === DeploymentAppType.helm && appDetails.appType === AppType.DEVTRON_APP
-                    ? `${appDetails.appName}-${appDetails.environmentName}`
-                    : appDetails.appName,
-            ),
-            k8sRequest: {
-                resourceIdentifier: {
-                    groupVersionKind: {
-                        Group: nodeDetails.group,
-                        Version: nodeDetails.version,
-                        Kind: nodeDetails.kind,
-                    },
-                    namespace: nodeDetails.namespace,
-                    name: nodeDetails.name,
+    // removed argocd server api dependencies and routed through k8s methods
+    const data = {
+        appId: appDetails.deploymentAppType === DeploymentAppType.argo_cd ? '' : getAppId(
+            appDetails.clusterId,
+            appDetails.namespace,
+            appDetails.deploymentAppType === DeploymentAppType.helm && appDetails.appType === AppType.DEVTRON_APP
+                ? `${appDetails.appName}-${appDetails.environmentName}`
+                : appDetails.appName,
+        ),
+        k8sRequest: {
+            resourceIdentifier: {
+                groupVersionKind: {
+                    Group: nodeDetails.group,
+                    Version: nodeDetails.version,
+                    Kind: nodeDetails.kind,
                 },
+                namespace: nodeDetails.namespace,
+                name: nodeDetails.name,
             },
-        }
-        return post(Routes.DELETE_RESOURCE, data)
+        },
+        ClusterId:  appDetails.deploymentAppType === DeploymentAppType.argo_cd ? appDetails.clusterId : null
```

Comment:
In the else part null is the required use case in the absence of argocd?

Reply:
clusterId is supplied in case of argocd and appId is required for helm releases